BitwardenShared/UI/Auth/AuthCoordinator.swift

```diff
@@ -846,15 +850,23 @@ final class AuthCoordinator: NSObject, // swiftlint:disable:this type_body_lengt
 
     /// Shows the vault unlock setup screen.
     ///
-    func showVaultUnlockSetup() {
+    /// - Parameter accountSetupFlow: The account setup flow that the user is in.
+    ///
+    func showVaultUnlockSetup(accountSetupFlow: AccountSetupFlow) {
         let processor = VaultUnlockSetupProcessor(
             coordinator: asAnyCoordinator(),
             services: services,
-            state: VaultUnlockSetupState(),
+            state: VaultUnlockSetupState(accountSetupFlow: accountSetupFlow),
             vaultUnlockSetupHelper: DefaultVaultUnlockSetupHelper(services: services)
         )
         let view = VaultUnlockSetupView(store: Store(processor: processor))
-        stackNavigator?.replace(view)
+        switch accountSetupFlow {
+        case .createAccount:
+            stackNavigator?.replace(view)
+        case .settings:
+            let viewController = UIHostingController(rootView: view)
+            stackNavigator?.push(viewController, navigationTitle: Localizations.setUpUnlock)
```

Comment:
should we add a test to make sure this push has the correct view being presented?

Reply:
Yeah, good catch!